context/app/static/js/components/entity-search/MetadataMenu/MetadataMenu.jsx

```diff
@@ -38,16 +39,34 @@ function NotebookMenuItem(props) {
   );
 }
 
-function MetadataMenu({ entityType }) {
+function MetadataMenu({ entityType, results }) {
   const lcPluralType = `${entityType.toLowerCase()}s`;
   const { selectedHits, createNotebook, closeMenu } = useMetadataMenu(lcPluralType);
   const menuID = 'metadata-menu';
 
   const { selectedRows } = useStore();
-  const errorMessage =
-    selectedRows.size > 10
-      ? `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`
-      : null;
+
+  // array that contains all the datasets from current pagination where mapped_data_access_level === 'Protected'.
+  const filteredDataset = results?.hits?.items?.filter(
+    (item) => item?.fields?.mapped_data_access_level === 'Protected',
+  );
+
+  const containsProtectedDataset = filteredDataset?.some((item) => {
+    const filteredDatasetIds = item?.id;
+    // spreading out the selectedRows Set to convert to an array then checking to see if any contains mapped_data_access_level === 'Protected'.
+    return [...selectedRows].some((selectedRowId) => filteredDatasetIds.includes(selectedRowId));
+  });
```

Comment:
Do the `results` only reflect the datasets shown on the current page? If so, we wouldn't be able to detect protected datasets selected on a different page of results. I think we need to move towards making a request to get the `protected_access_level` for the datasets in `selectedHits`.

Reply:
We could use an [aggs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html) query to avoid having to loop over the results that would be returned from this new request.

Reply:
Done